### ISSUE TITLE
Improve plugin behaviour during upgrades

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#7308]
Our current recommendation for upgrades is that extra plugins not be present when running the upgrade script, and for them to be (re)introduced only afterwards.

In preparation for https://github.com/pkp/pkp-lib/issues/4679, and to simplify upgrades in general, it would be nice to be able to trust plugins to behave in place during the upgrade.

There is already a `RUNNING_UPGRADE` constant defined when the upgrade script is running; the contract with plugins around this constant should be clarified and plugin behaviour checked over to ensure that we can avoid gotchas.

**Comment by jonasraoni** [comment]
The constant is good for the submodule plugins, but for the external ones there are also the breaking changes that come together with the PHP version bump.
Perhaps this issue can be closed.

**Comment by asmecher** [comment]
Another relatively quick option could be to have plugins self-describe their own compatibility, so that OJS/OMP/OPS can run a quick check before attempting to register. If the compatibility assertion was in code (as opposed to e.g. `version.xml`) then it would be very performant. I'd like to leave this (and related issues) option for possible sprint work.

**Comment by jonasraoni** [comment]
Yeah, there are issues for that (I've tagged Erik on the 7531, as he was doing coding something for that on the Turin sprint):
https://github.com/pkp/pkp-lib/issues/7531
https://github.com/pkp/pkp-lib/issues/9297
https://github.com/pkp/pkp-lib/discussions/8017

**Comment by asmecher** [comment]
Yup, those capture it well enough. I agree, this one can be closed.
